core/src/test/java/org/apache/calcite/test/RelOptRulesTest.java

```diff
@@ -2699,6 +2699,35 @@ private void checkPushJoinThroughUnionOnRightDoesNotMatchSemiOrAntiJoin(JoinRelT
         .check();
   }
 
+  /** Tests {@link org.apache.calcite.rel.rules.MinusToDistinctRule},
```

Comment:
Could you add an E2E test case using quidem, to check the final result correct or not?

Reply:
+1, also wondering

Reply:
Yes,I'm looking for a single test with the correctness of optimizes rule

Reply:
> +1, also wondering

@herunkang2018 @tanclary ,I couldn't find a case where quidem was used to test the correctness of the optimization rules,could you give me a unit test or class to refer to? Thank you very much.

Reply:
I looked through and I also could not find a good example, I don't think that should hold up this PR.

Reply:
@LakeShen I think we chould add the test in [set-op.iq](https://github.com/apache/calcite/blob/2a96512c352bda4a5d9c0c80730f5c115ac363d6/core/src/test/resources/sql/set-op.iq#L157).

Reply:
In set-op.iq we need to support using this Rule to work. As far as I know It didn't. But  we can try to read 
 `https://github.com/apache/calcite/blob/main/core/src/test/java/org/apache/calcite/test/JdbcTest.java#L3749C14-L3749C27`. @LakeShen Can we add tests like this?

Reply:
In set-op.iq we need to support using this Rule to work. As far as I know It didn't. But  we can try to read 
 `https://github.com/apache/calcite/blob/main/core/src/test/java/org/apache/calcite/test/JdbcTest.java#L3749C14-L3749C27`. @LakeShen Can we add tests like this?

Reply:
@NobiGo @herunkang2018 ,thank you very much for your advice,I will take it later.

Reply:
> In set-op.iq we need to support using this Rule to work. As far as I know It didn't. But we can try to read `https://github.com/apache/calcite/blob/main/core/src/test/java/org/apache/calcite/test/JdbcTest.java#L3749C14-L3749C27`. @LakeShen Can we add tests like this?

Hi @NobiGo ,I have added the test for this rule in JdbcTest,please help me to review again, thank you very much

Reply:
We are abusing `JdbcTest` but there seems to be no better way at the moment (https://lists.apache.org/thread/r4yhn77m92fodmz8xm6k40sfd130w7hh), and it's definitely better to have a test here than no test at all.